pkg/kubelet/dockertools/runner.go

```diff
@@ -0,0 +1,149 @@
+			CPUShares:    milliCPUToShares(container.Resources.Limits.Cpu().MilliValue()),
+			WorkingDir:   container.WorkingDir,
+		},
+	}
+
+	glog.Infof("Setting entrypoint and cmd for %v/%v/%v", pod.Namespace, pod.Name, container.Name)
+	setEntrypointAndCommand(container, &dockerOpts)
+
+	dockerContainer, err := r.Client.CreateContainer(dockerOpts)
+	if err != nil {
+		if ref != nil {
+			r.Recorder.Eventf(ref, "failed", "Failed to create docker container with error: %v", err)
+		}
+		return "", err
+	}
+
+	if ref != nil {
+		r.Recorder.Eventf(ref, "created", "Created with docker id %v", dockerContainer.ID)
+	}
+
+	// The reason we create and mount the log file in here (not in kubelet) is because
+	// the file's location depends on the ID of the container, and we need to create and
+	// mount the file before actually starting the container.
+	// TODO(yifan): Consider to pull this logic out since we might need to reuse it in
+	// other container runtime.
+	if opts.PodContainerDir != "" && len(container.TerminationMessagePath) != 0 {
+		containerLogPath := path.Join(opts.PodContainerDir, dockerContainer.ID)
+		fs, err := os.Create(containerLogPath)
+		if err != nil {
+			// TODO: Clean up the previouly created dir? return the error?
+			glog.Errorf("Error on creating termination-log file %q: %v", containerLogPath, err)
+		} else {
+			fs.Close() // Close immediately; we're just doing a `touch` here
+			b := fmt.Sprintf("%s:%s", containerLogPath, container.TerminationMessagePath)
+			opts.Binds = append(opts.Binds, b)
+		}
+	}
+
+	privileged := false
+	if capabilities.Get().AllowPrivileged {
+		privileged = container.Privileged
+	} else if container.Privileged {
+		return "", fmt.Errorf("container requested privileged mode, but it is disallowed globally.")
+	}
+
+	capAdd, capDrop := makeCapabilites(container.Capabilities.Add, container.Capabilities.Drop)
+	hc := &docker.HostConfig{
+		PortBindings: portBindings,
+		Binds:        opts.Binds,
+		NetworkMode:  opts.NetMode,
+		IpcMode:      opts.IpcMode,
+		Privileged:   privileged,
+		CapAdd:       capAdd,
+		CapDrop:      capDrop,
+	}
+	if len(opts.DNS) > 0 {
+		hc.DNS = opts.DNS
+	}
+	if len(opts.DNSSearch) > 0 {
+		hc.DNSSearch = opts.DNSSearch
+	}
+
+	if err = r.Client.StartContainer(dockerContainer.ID, hc); err != nil {
+		if ref != nil {
+			r.Recorder.Eventf(ref, "failed",
+				"Failed to start with docker id %v with error: %v", dockerContainer.ID, err)
+		}
+		return "", err
+	}
+	if ref != nil {
+		r.Recorder.Eventf(ref, "started", "Started with docker id %v", dockerContainer.ID)
+	}
+	return dockerContainer.ID, nil
+}
+
+func setEntrypointAndCommand(container *api.Container, opts *docker.CreateContainerOptions) {
+	if len(container.Command) != 0 {
+		glog.Infof("Setting entrypoint to: %v", container.Command)
+		opts.Config.Entrypoint = container.Command
+	}
+	if len(container.Args) != 0 {
+		glog.Infof("Setting cmd to: %v", container.Args)
```

Comment:
@vmarmol yep those are for debugging, i will take them out
On Mon, Mar 30, 2015 at 11:59 PM Victor Marmol <notifications@github.com>
wrote:

> In pkg/kubelet/dockertools/runner.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6100#discussion_r27451543>
> :
>
> > +		}
> > +		return "", err
> > +	}
> > +	if ref != nil {
> > +		r.Recorder.Eventf(ref, "started", "Started with docker id %v", dockerContainer.ID)
> > +	}
> > +	return dockerContainer.ID, nil
> > +}
> > +
> > +func setEntrypointAndCommand(container *api.Container, opts *docker.
>
> CreateContainerOptions) {
> > +	if len(container.Command) != 0 {
> > +		glog.Infof("Setting entrypoint to: %v", container.Command)
> > +		opts.Config.Entrypoint = container.Command
> > +	}
> > +	if len(container.Args) != 0 {
> > +		glog.Infof("Setting cmd to: %v", container.Args)
>
> Same :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6100/files#r27451543>
> .
>